src/client.ts

```diff
@@ -163,4 +169,39 @@ export class CortexClient {
   async listCatalogs() {
     return Catalog.list(this.apiClient);
   }
+
+  async createWebScraperIndexer(
+    name: string,
+    catalogName: string,
+    schedule: IndexerScheduleFrequency,
+    config: WebScraperDataSourceConfig,
+  ): Promise<Indexer> {
+    return Indexer.create(this.apiClient, name, catalogName, schedule, {
+      type: "webScraper",
+      config,
+    });
+  }
+
+  async createGithubIndexer(
+    name: string,
+    catalogName: string,
+    schedule: IndexerScheduleFrequency,
+    config: GithubDataSourceConfig,
+  ): Promise<Indexer> {
+    if (!config.code) {
+      config.code = {};
+    }
+    return Indexer.create(this.apiClient, name, catalogName, schedule, {
+      type: "github",
+      config,
+    });
+  }
```

Comment:
Just thinking about the API pattern we are subscribing ourselves to. If we add more indexer types, do we add more helpers for this?

Would it be a simpler API to have a `createIndexer` method that takes a type?

```
async createIndexer<Type extends 'webScraper' | 'github'>(name: string, catalogName: string, type: Type, config: Type extends 'webScraper' ? WebScraperDataSourceConfig : GithubDataSourceConfig): Promise<Indexer> {
  return Indexer.create(this.apiClient, name, catalogName, schedule, {
      type,
      config,
    });
}
```

I'm not married to this idea, but wanted to check that we had considered it.

Reply:
I did consider a generic `createIndexer` method but I think it will be less usable than the per-type factory methods, and usability is more important than the slight maintenance overhead for us. 

However, I do share the concern that this approach will eventually pollute `CortexClient` with too many methods. Instead, we could add these static methods to `Indexer` class instead, make them take `CortexClient` as the first argument, but for that to work, we'll need to make `CortexClient.apiClient` public or package-internal (if TS supports that), which I think is a good idea anyway because it will make it easier for people to call our REST APIs that haven't been exposed in the SDK yet. 
I didn't do it because it would be a break with the existing pattern of defining creation methods on the client.  
What do you think about that? 
Also interested in @EvanBoyle's thoughts

Reply:
Is indexer creation from the SDK useful?

Reply:
Given our current customer profile, most will use the App for creating indexers (although that holds for catalogs as well). Once you have bigger customers and need to be a part of some workflows (especially things like app deployment), ability to do it programmatically will be a must. For example, tons of Azure Search customers managed indexers programmatically. 

Since it's indexer management won't be a common task for the foreseeable future, this looks like another argument for moving these APIs to `Indexer` class.

Reply:
I should've also shared my motivation for doing this now rather than later: I want to experiment with writing API documentation, and since we don't have REST API documentation, I want to write SDK documentation for hosted indexers. In order to do that, indexers need to be in the SDK :)